machetli/pddl/visitors.py

```diff
@@ -310,3 +310,134 @@ def visit_condition_atom(self, atom) -> Atom:
+
+
+class TaskElementEraseInitFactVisitor(TaskElementVisitor):
+    """Partial implementation of TaskElementVisitor interface for deletion of fact from init."""
+
+    def __init__(self, fact):
+        self.the_fact = fact
+
+    def visit_task(self, task):
+        new_init = [atom for atom in task.init if isinstance(atom, Assign) or atom != self.the_fact]
+
+        return Task(task.domain_name, task.task_name, task.requirements, task.types, task.objects, task.predicates,
+                    task.functions, new_init, task.goal, task.actions, task.axioms, task.use_min_cost_metric)
+
+
+class TaskElementEraseGoalLiteralVisitor(TaskElementVisitor):
+    """Partial implementation of TaskElementVisitor interface for deleting a literal from the goal."""
+
+    def __init__(self, literal):
+        self.the_literal = literal
+
+    def visit_task(self, task):
+        new_goal = task.goal.accept(self)
+
+        return Task(task.domain_name, task.task_name, task.requirements, task.types, task.objects, task.predicates,
+                    task.functions, task.init, new_goal, task.actions, task.axioms, task.use_min_cost_metric)
+
+    def visit_condition_falsity(self, falsity) -> Falsity:
+        return Falsity()
+
+    def visit_condition_truth(self, truth) -> Truth:
+        return Truth()
+
+    def visit_condition_conjunction(self, conjunction) -> Conjunction:
+        new_parts = []
+        for part in conjunction.parts:
+            new_parts.append(part.accept(self))
+        new_parts = [part for part in new_parts if part is not None]
+        return Conjunction(new_parts).simplified()
+
+    def visit_condition_disjunction(self, disjunction) -> Disjunction:
+        new_parts = []
+        for part in disjunction.parts:
+            new_parts.append(part.accept(self))
+        new_parts = [part for part in new_parts if part is not None]
+        return Disjunction(new_parts).simplified()
+
+    def visit_condition_universal(self, universal_condition) -> UniversalCondition:
+        new_parts = []
+        for part in universal_condition.parts:
+            new_parts.append(part.accept(self))
+        new_parts = [part for part in new_parts if part is not None]
+        return UniversalCondition(universal_condition.parameters, new_parts).simplified()
+
+    def visit_condition_existential(self, existential_condition) -> ExistentialCondition:
+        new_parts = []
+        for part in existential_condition.parts:
+            new_parts.append(part.accept(self))
+        new_parts = [part for part in new_parts if part is not None]
+        return ExistentialCondition(existential_condition.parameters, new_parts).simplified()
+
+    def visit_condition_atom(self, atom) -> Atom:
+        return Truth() if atom == self.the_literal else atom
+
+    def visit_condition_negated_atom(self, negated_atom) -> NegatedAtom:
+        return Truth() if atom == self.the_literal else negated_atom 
+
+
+class GetLiteralsVisitor:
```

Comment:
I'm not sure a visitor makes sense here. Wouldn't a recursive function be a better fit?